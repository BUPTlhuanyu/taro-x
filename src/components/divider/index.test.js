/*
 * @Descripttion: 
 * @Author: lhuanyu
 * @Date: 2020-03-24 16:56:41
 * @LastEditors: lhuanyu
 * @LastEditTime: 2020-03-24 17:55:59
 */
import Nerv, { findDOMNode } from 'nervjs'
import { renderToString } from 'nerv-server'
import { Simulate, renderIntoDocument } from 'nerv-test-utils'

import AtDivider from '../../../.temp/components/divider/index'

describe('AtDivider Snap', () => {
  it('render AtDivider -- props content(分割线)', () => {
    const component = renderToString(<AtDivider content = '分割线' />)
    expect(component).toMatchSnapshot()
  })
})


describe('AtDivider Event', () => {
  it('AtDivider click', () => {
    const onClick = jest.fn()
    
    const component = renderIntoDocument(<AtDivider content = '分割线' onClick = {onClick} fontColor='#E3E6E9' fontSize="50"/>)
    const componentDom = findDOMNode(component, 'at-divider')

    const contentDom = componentDom.querySelector('.at-divider__content');

    Simulate.click(contentDom)
    expect(onClick).toHaveBeenCalledTimes(1)
  })
})