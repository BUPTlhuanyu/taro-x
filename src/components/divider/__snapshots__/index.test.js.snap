// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtDivider Snap render AtDivider -- props content(分割线) 1`] = `"<div style=\\"height:;\\" class=\\"at-divider\\"><div style=\\"color:;font-size:;\\" class=\\"at-divider__content\\">分割线</div><div style=\\"border-top-color:;border-top-style:solid;\\" class=\\"at-divider__line\\"></div></div>"`;
